From 861020fd667420b6367b0e4df7f9141d14503cc9 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Tue, 3 Jul 2018 21:29:54 +0300
Subject: [PATCH 2/2] power: Allow devices disabling power stats completely

* On some devices even attempting to read the stats nodes results
  in kernel lockup and reboot from watchdog bite.
  This has been noticed on a lot of msm8916 devices, so add a way
  to disable stats completely in order to allow using the binderized
  power HAL and get rid of the old one from device/qcom/common

Change-Id: I134c40548dd883aaf72cf0c212be978305366521
---
 Android.mk | 4 ++++
 Power.cpp  | 8 ++++++++
 2 files changed, 12 insertions(+)

diff --git a/Android.mk b/Android.mk
index 09d659a..2a24ad6 100644
--- a/Android.mk
+++ b/Android.mk
@@ -134,6 +134,10 @@ ifeq ($(TARGET_HAS_LEGACY_POWER_STATS),true)
     LOCAL_CFLAGS += -DLEGACY_STATS
 endif
 
+ifeq ($(TARGET_HAS_NO_POWER_STATS),true)
+    LOCAL_CFLAGS += -DNO_STATS
+endif
+
 ifneq ($(TARGET_RPM_STAT),)
     LOCAL_CFLAGS += -DRPM_STAT=\"$(TARGET_RPM_STAT)\"
 endif
diff --git a/Power.cpp b/Power.cpp
index f857041..a359d7a 100644
--- a/Power.cpp
+++ b/Power.cpp
@@ -74,6 +74,13 @@ Return<void> Power::setFeature(Feature feature, bool activate)  {
     return Void();
 }
 
+#ifdef NO_STATS
+Return<void> Power::getPlatformLowPowerStats(getPlatformLowPowerStats_cb _hidl_cb) {
+    hidl_vec<PowerStatePlatformSleepState> states;
+    states.resize(0);
+    _hidl_cb(states, Status::SUCCESS);
+    return Void();
+#else
 Return<void> Power::getPlatformLowPowerStats(getPlatformLowPowerStats_cb _hidl_cb) {
     hidl_vec<PowerStatePlatformSleepState> states;
     uint64_t stats[MAX_PLATFORM_STATS * MAX_RPM_PARAMS] = {0};
@@ -169,6 +176,7 @@ done:
     _hidl_cb(states, Status::SUCCESS);
     return Void();
 }
+#endif
 
 #ifndef V1_0_HAL
 // Methods from ::android::hardware::power::V1_1::IPower follow.
-- 
2.17.1

